operator/v1/types_console.go

```diff
@@ -143,8 +143,94 @@ type Capability struct {
 	Visibility CapabilityVisibility `json:"visibility"`
 }
 
+// ThemeType is the value of the CustomLogo Theme Type field that determines the theme mode in the console UI.
+type ThemeType string
+
+// ThemeType values
+const (
+	// ThemeTypeDark represents the dark theme for the console UI.
+	ThemeTypeDark ThemeType = "Dark"
+
+	// ThemeTypeLight represents the light theme for the console UI.
+	ThemeTypeLight ThemeType = "Light"
+
+	// ThemeTypeDefault represents the default theme for all unspecified themes.
+	ThemeTypeDefault ThemeType = "Default"
+)
+
+// LogoType is the value of the CustomLogo Type field that determines if the logo is for the masthead or the favicon in the console UI.
+type LogoType string
+
+const (
+	// Masthead represents the logo in the masthead.
+	LogoTypeMasthead LogoType = "Masthead"
+
+	// Favicon represents the favicon logo.
+	LogoTypeFavicon LogoType = "Favicon"
+)
+
+// Theme defines a theme type for the console UI.
+type Theme struct {
+	// type is used to specify what theme(s) a logo will apply to in the console UI.
+	// type is a required field that allows values of Dark, Light, and Default.
+	// When set to Dark, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Dark theme.
+	// When set to Light, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Light theme.
+	// When set to Default, the logo file referenced in the 'file' field will be used for any themes types that do not have an explicit configuration.
+	// +kubebuilder:validation:Enum="Dark";"Light";"Default"
+	// +required
+	Type ThemeType `json:"type"`
+
+	// file is used by the console to locate the specified file containing a custom logo.
+	// file is a required field that references a ConfigMap name and key that contains the custom logo file in the openshift-config namespace.
+	// You can create it with a command like:
+	// - 'oc create configmap custom-logos-config --namespace=openshift-config --from-file=/path/to/file'
+	// The ConfigMap key must include the file extension for the dark, light, or default theme so that the console serves the file with the correct MIME type, which can be '.svg', '.png', or whatever the browser supports.
+	// Recommended Masthead and About Modal logo specifications:
```

Comment:
What is about modal? Do we need to add the about modal part here since it's explained elsewhere?

Here it read to me as if that was a different option I could choose in the type enum

Reply:
About modal displays info about cluster, channels, subscription... IN the modal we are also rendering the logo in the top left corner.
<img width="1830" alt="Screenshot 2025-03-06 at 14 57 57" src="https://github.com/user-attachments/assets/5f3cd97f-3bf2-4bad-b0d3-61af91540a24" />

Reply:
Ok, so somewhere we want to document that the Masthead logo is also used for the About modal? (I think we already have that) I'm not sure we need to mention it here specifically